### PR TITLE
docs(development): adding new controller explanation

### PR DESCRIPTION
**Pull request by surajssd**
Fixes partially https://github.com/kedgeproject/kedge/issues/62

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -504,3 +504,50 @@ Issue is a duplicate of other already existing issue.
 #### [invalid](https://github.com/kedgeproject/kedge/labels/invalid)
 
 #### [wontfix](https://github.com/kedgeproject/kedge/labels/wontfix)
+
+
+## Code instructions
```
Comment (kadel) [inline docs/development.md]: I would move it up, before the Labeling guide section

File: docs/development.md [hunk 2]
```diff
@@ -504,3 +504,50 @@ Issue is a duplicate of other already existing issue.
...
+
+### Things to take care of when adding new type:
+
+- Creating type struct
+
+  - Make a struct type named `ControllerSpecMod` and add it to [types.go](https://github.com/kedgeproject/kedge/blob/cd5297cfdbd2f1daa510824d49c9d3d649ffc0b8/pkg/spec/types.go).
```
Comment (kadel) [inline docs/development.md]: Maybe mentioning that it should contain ObjectMeta?

File: docs/development.md [hunk 3]
```diff
@@ -504,3 +504,50 @@ Issue is a duplicate of other already existing issue.
...
+- Interface's Fix method
+
+  - This is where we can do most of fixing of user provided data, we can do
+  auto-population of information that user has not given us.
+  - Make sure to call the [`ControllerFields.fixControllerFields()`](https://github.com/surajssd/kedge/blob/9b732e3d526b03b197c9ba623627099f221f023c/pkg/spec/resources.go#L183).
+  - Now after that add any Controller specific Fix methods.
+  - [Sample Fix method](https://github.com/kedgeproject/kedge/blob/2f1cf4ee6a90e5f0911ac8d8fcee0d751fc97fa8/pkg/spec/job.go#L42)
+  of Job controller.
+
+- Interface's Trasform method
```
Comment (kadel) [inline docs/development.md]: missing 'n' in Transform

File: docs/development.md (hunk 1 above)
Comment (surajssd) [reply docs/development.md]: done!

File: docs/development.md (hunk 3 above)
Comment (surajssd) [reply docs/development.md]: done!

File: docs/development.md (hunk 2 above)
Comment (surajssd) [reply docs/development.md]: ObjectMeta is already present in `ControllerFields`, so when doing controller types we don't need it additionally!

**Comment by surajssd** [comment]
@cdrage @containscafeine can you give it a look and suggest changes if need be?

**Comment by cdrage** [comment]
LGTM

**Pull request merged**